tests/integration/conftest.py

```diff
@@ -512,18 +549,18 @@ def required_packages(shell):
     """
     try:
         required_packages = os.environ.get("TEST_REQUIRES").split(" ")
-        for package in required_packages:
-            print(f"\nPREPARE: Installing required {package}")
-            assert shell(f"yum install -y {package}")
+    except KeyError:
+        raise
 
-        yield
+    for package in required_packages:
+        print(f"\nPREPARE: Installing required {package}")
+        assert shell(f"yum install -y {package}")
 
-        for package in required_packages:
-            print(f"\nCLEANUP: Removing previously installed required {package}")
-            assert shell(f"yum remove -y *{package}*")
+    yield
 
-    except KeyError:
-        raise
+    for package in required_packages:
+        print(f"\nCLEANUP: Removing previously installed required {package}")
+        assert shell(f"yum remove -y *{package}*")
```

Comment:
This might be a good opportunity to drop this (`required_packages()`) in favor of https://tmt.readthedocs.io/en/stable/spec/tests.html#require
Honestly, I'm not sure if tmt will handle the teardown part and remove the package after each test, or if it will be able to handle different sets of packages for different distributions.
Looking in the docs we might need to use prepare to be able to run with different sets of packages, which might not be viable for our current set up.
I'll leave that up to you if you want to pursue this in either this or another PR.